### PR TITLE
docker.service: don't limit tasks

### PR DESCRIPTION
**Pull request by pcarrier**
Default is 512 tasks.

### CODE REVIEW

**Comment by thaJeztah** [comment]
Actually, we had this change, but it caused issues on older kernels, so was removed again; https://github.com/docker/docker/pull/20167

**Comment by pcarrier** [comment]
And nobody thought to leave it as a comment??

**Comment by calavera** [comment]
Let's leave it commented with an explanation, so people can uncomment it themselves:

```
# Uncomment TasksMax if your systemd version supports it.
# Only systemd 226 and avobe support this version.
#TasksMax=infinity
```

**Comment by pcarrier** [comment]
Should we go the other way around and leave it uncommented by default,
given the failure scenario of "doesn't start" is much more discoverable and
less likely to create surprises in production for users with _any_ kind of
QA?

Pierre Carrier

**Comment by thaJeztah** [comment]
I don't think we should; when installing docker for production, there's more things that need to be tuned/configured (e.g. properly setting up devicemapper)

We may need to change that comment though; because people shouldn't edit the default docker.service, but [use a drop-in file](https://docs.docker.com/engine/admin/systemd/#custom-docker-daemon-options)

**Comment by pcarrier** [comment]
Updated with the last comment suggestion; feel free to suggest something different or open a separate PR and close this one at your convenience :)

File: contrib/init/systemd/docker.service [hunk 1]
```diff
@@ -14,6 +14,9 @@ MountFlags=slave
 LimitNOFILE=1048576
 LimitNPROC=1048576
 LimitCORE=infinity
+# Uncomment TasksMax if your systemd version supports it.
+# Only systemd 226 and avobe support this version.
```
Comment (cyphar) [inline contrib/init/systemd/docker.service]: s/avobe/above/

**Comment by pcarrier** [comment]
@thaJeztah @cyphar anything else?

**Comment by vdemeester** [comment]
A weak LGTM for me, because of what @thaJeztah said. It cannot hurt to have a comment in there though but I think we might want to document this option in the systemd article still.

> We may need to change that comment though; because people shouldn't edit the default docker.service, but [use a drop-in file](https://docs.docker.com/engine/admin/systemd/#custom-docker-daemon-options)

**Comment by pcarrier** [comment]
When you include things like:

```
LimitNOFILE=1048576
LimitNPROC=1048576
```

I assume this file is ready to consume, not some minimal demo.

So if it's not, I think a comment is in order.

FWIW the impact for us was http://status.meteor.com/incidents/0nx4lggyggcp

**Comment by clnperez** [comment]
@pcarrier - I just submitted a PR based on the comment idea here that might avoid what you guys ran into in that incident. I'm happy to take out the change to the service file itself if docker is willing to accept both PRs. See PR #21628

**Comment by pcarrier** [comment]
@clnperez we install docker without going through a deb, but SGTM ;)

**Comment by calavera** [comment]
LGTM

**Comment by vdemeester** [comment]
LGTM :smile:

**Pull request merged**